app/src/main/java/com/hover/stax/transfers/TransferFragment.kt

```diff
@@ -227,7 +235,10 @@ class TransferFragment : AbstractFormFragment(), ActionSelect.HighlightListener
         val recipientError = transferViewModel.recipientErrors(actionSelectViewModel.activeAction.value)
         contactInput.setState(recipientError, if (recipientError == null) AbstractStatefulInput.SUCCESS else AbstractStatefulInput.ERROR)
 
-        return channelError == null && actionError == null && amountError == null && recipientError == null
+        val nonTemplateVariableHasError = transferViewModel.nonTemplateVariablesAnError()
```

Comment:
As noted elsewhere, if you move this validation call from the viewmodel to the adapter it will simplify the logic some